presentation/src/main/java/com/smile/presentation/MainActivityFragment.kt

```diff
@@ -53,27 +53,18 @@ class MainActivityFragment : BaseFragment() {
         recyclerView.setHasFixedSize(true)
         recyclerView.adapter = characterAdapter
         populateData()
+        errorHandler()
     }
 
     private fun populateData() {
-        viewModel.characterLiveData.observe(this,
-            object : Kobserver<List<CharacterUI>>() {
-
-                override fun onException(exception: Exception) {
-                    super.onException(exception)
-                    val msg = exception.message ?: getString(R.string.error_generic)
-                    Toast.makeText(context, msg, Toast.LENGTH_LONG)
-                        .show()
-                }
-
-                override fun onSuccess(data: List<CharacterUI>) {
-                    characterItems.addAll(data)
-                    characterAdapter?.notifyDataSetChanged()
-                }
-
-            })
+        viewModel.characterLiveData.observe(this, Observer {
+            characterItems.addAll(it)
+            characterAdapter?.notifyDataSetChanged()
```

Comment:
Nit: If you are positive this will always be initialized when the fragment is created, you can use `lateinit`.

Reply:
@chizoba I tend to avoid `lateinit` though, it feels too hacky for me

Reply:
This is how i use it:
- If i never have to verify if the property is available or not, I use it.
- If i need to verify if the property is available or not, i use nullable type